### PR TITLE
fix: fix launcher api retry n timeouts

### PR DESCRIPTION
**Pull request by kkisic**
## Context
Frequent timeouts from Launcher to API.

## Objective
Switch default http client to retryablehttp client and add timeout, retry configurations.

## References
https://github.com/screwdriver-cd/log-service/pull/33

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by parthasl** [comment]
@kkisic Started the review but unfortunately didn't submit it, realized it after this PR is merged.

File: screwdriver/screwdriver.go [hunk 1]
```diff
@@ -177,116 +200,101 @@ func tokenHeader(token string) string {
...
-	return retry(maxAttempts, func() ([]byte, error) {
-		attemptNumber++
-		res, err = a.client.Do(req)
-		if err != nil {
-			log.Printf("WARNING: received error from GET(%s): %v "+
-				"(attempt %d of %d)", url.String(), err, attemptNumber, maxAttempts)
-			return nil, err
+	if res.StatusCode/100 != 2 {
+		var errParse SDError
+		parseError := json.Unmarshal(body, &errParse)
```
[the remark this answers is not in the document]
Comment (kkisic) [reply screwdriver/screwdriver.go:233]: It's the same reason for this.
https://github.com/screwdriver-cd/log-service/pull/34#discussion_r445239833

File: screwdriver/screwdriver.go [hunk 2]
```diff
@@ -177,116 +200,101 @@ func tokenHeader(token string) string {
...
+	}
 
-	return retry(maxAttempts, func() ([]byte, error) {
-		attemptNumber++
-		res, err = a.client.Do(req)
-		if err != nil {
-			log.Printf("WARNING: received error from GET(%s): %v "+
-				"(attempt %d of %d)", url.String(), err, attemptNumber, maxAttempts)
-			return nil, err
+	if res.StatusCode/100 != 2 {
```
[the remark this answers is not in the document]
Comment (kkisic) [reply screwdriver/screwdriver.go:231]: I think we need to check after reading body to parse error response if the status code is not 2XX.

[the remark this answers is not in the document]
Comment (parthasl) [reply screwdriver/screwdriver.go:231]: parsed error response used?

[the remark this answers is not in the document]
Comment (kkisic) [reply screwdriver/screwdriver.go:231]: Certainly we don't use error response, but we can confirm whether or not JSON of the error response is correct format.

**Pull request merged**